### PR TITLE
Change allow superuser to edit future posts

### PR DESCRIPTION
**Pull request by ikonitas**
Fixes https://github.com/blancltd/django-glitter-news/issues/2

### CODE REVIEW

File: glitter_news/views.py [hunk 1]
```diff
@@ -44,6 +44,12 @@ class PostDetailView(GlitterDetailMixin, DateDetailView):
     month_format = '%m'
     date_field = 'date'
 
+    def get_allow_future(self):
+        """ Only superusers can edit future posts. """
+        if self.request.user.is_superuser:
```
Comment (tomkins) [inline glitter_news/views.py]: Think we need to open this to users other than superusers.

Can we change this to something like:

``` python
self.request.user.has_perm('glitter_news.edit_post')
```

Or:

``` python
qs = self.get_queryset()
self.request.user.has_perm(
    '{}.edit_{}'.format(qs.model._meta.app_label, qs.model._meta.model_name
)
```

Up to you (and test the second one, I think that's the right format for it).

**Review by tomkins — CHANGES_REQUESTED** [review]
Will be nice to get this fixed!

**Comment by ikonitas** [comment]
Updated.

File: glitter_news/views.py [hunk 2]
```diff
@@ -44,6 +44,18 @@ class PostDetailView(GlitterDetailMixin, DateDetailView):
...
 
+    def get_allow_future(self):
+        """
+        Only superusers and users with the permission can edit the post.
+        """
+        qs = self.get_queryset()
+        post_edit_permission = '{}.edit_{}'.format(
+            qs.model._meta.app_label, qs.model._meta.model_name
+        )
+        if self.request.user.is_superuser or self.request.user.has_perm(post_edit_permission):
```
Comment (tomkins) [inline glitter_news/views.py]: Do we need the additional check for a superuser? As has_perm will always return True for superusers.

Up to you!

**Review by tomkins — APPROVED** [review]
:shipit: if you want, tweak if you feel it's needed.

**Comment by ikonitas** [comment]
Removed we don't need useless code! :+1:

**Comment by tomkins** [comment]
Super - :shipit:!

**Pull request merged**